set.mm

```diff
@@ -215251,6 +215253,65 @@ is an important property of monoids (see ~ mndid ), and therefore also for
       IUGUJUKULUMUNAHUOTUPUSVEACVGUQTUR $.
   $}
 
+  ${
+    $d n u v w x y z B $.  $d n u v w x y z O $.  $d n u v w x y z ph $.
+    $d u v w y z N $.  $d n u v w x y z .+ $.  $d u v w y z X $.
+    $d u v w y ps $.
+    grprinvlem.c $e |- ( ( ph /\ x e. B /\ y e. B ) -> ( x .+ y ) e. B ) $.
+    grprinvlem.o $e |- ( ph -> O e. B ) $.
+    grprinvlem.i $e |- ( ( ph /\ x e. B ) -> ( O .+ x ) = x ) $.
+    grprinvlem.a $e |- ( ( ph /\ ( x e. B /\ y e. B /\ z e. B ) )
+          -> ( ( x .+ y ) .+ z ) = ( x .+ ( y .+ z ) ) ) $.
+    grprinvlem.n $e |- ( ( ph /\ x e. B ) -> E. y e. B ( y .+ x ) = O ) $.
+    ${
+      grprinvlem.x $e |- ( ( ph /\ ps ) -> X e. B ) $.
+      grprinvlem.e $e |- ( ( ph /\ ps ) -> ( X .+ X ) = X ) $.
+      $( Lemma for ~ grprinvd .  (Contributed by NM, 9-Aug-2013.) $)
+      grprinvlem $p |- ( ( ph /\ ps ) -> X = O ) $=
+        ( cv co wceq wcel vu vv vw wa wrex wral ralrimiva oveq2 rexbidv cbvralv
+        eqeq1d sylib syl2an2r oveq2d adantr simprr oveq1d w3a caovassg ad4ant14
+        rspccva simprl caovassd id eqeq12d rspcdva 3eqtr3d rexlimddv ) ABUDZDQZ
+        IGRZHSZIHSDFAVJEQZGRZHSZDFUEZEFUFZBIFTZVLDFUEZAVJCQZGRZHSZDFUEZCFUFVQAW
+        CCFNUGWCVPCEFVTVMSZWBVODFWDWAVNHVTVMVJGUHUKUIUJULOVPVSEIFVMISZVOVLDFWEV
+        NVKHVMIVJGUHUKUIVAUMVIVJFTZVLUDZUDZVJIIGRZGRZVKIHVIWJVKSWGVIWIIVJGPUNUO
+        WHVKIGRHIGRZWJIWHVKHIGVIWFVLUPZUQWHUAUBUCVJIIFGAUAQZFTUBQZFTUCQZFTURWMW
+        NGRWOGRWMWNWOGRGRSBWGACDEWMWNWOFGMUSUTVIWFVLVBVIVRWGOUOZWPVCVIWKISZWGVI
+        HVJGRZVJSZWQDFIVJISZWRWKVJIVJIHGUHWTVDVEAWSDFUFZBAHVTGRZVTSZCFUFXAAXCCF
+        LUGXCWSCDFVTVJSZXBWRVTVJVTVJHGUHXDVDVEUJULUOOVFUOVGWLVGVH $.
+    $}
+
+    ${
+      grprinvd.x $e |- ( ( ph /\ ps ) -> X e. B ) $.
+      grprinvd.n $e |- ( ( ph /\ ps ) -> N e. B ) $.
+      grprinvd.e $e |- ( ( ph /\ ps ) -> ( N .+ X ) = O ) $.
+      $( Deduce right inverse from left inverse and left identity in an
+         associative structure (such as a group).  (Contributed by NM,
+         10-Aug-2013.)  (Proof shortened by Mario Carneiro, 6-Jan-2015.) $)
+      grprinvd $p |- ( ( ph /\ ps ) -> ( X .+ N ) = O ) $=
+        ( co wcel vu vv vw wa cv caovclg adantlr caovcld wceq caovassg caovassd
+        3expb w3a oveq1d oveq2 id eqeq12d wral ralrimiva cbvralv adantr rspcdva
+        sylib 3eqtr3d oveq2d eqtrd grprinvlem ) ABCDEFGIJHGSZKLMNOABUDZUAUBJHFF
+        FGAUAUEZFTZUBUEZFTZUDVJVLGSZFTBACDVJVLFFFGACUEZFTDUEZFTVOVPGSFTKULUFUGP
+        QUHZVIVHVHGSJHVHGSZGSVHVIUAUBUCJHVHFGAVKVMUCUEZFTUMVNVSGSVJVLVSGSGSUIBA
+        CDEVJVLVSFGNUJUGZPQVQUKVIVRHJGVIHJGSZHGSIHGSZVRHVIWAIHGRUNVIUAUBUCHJHFG
+        VTQPQUKVIIVPGSZVPUIZWBHUIDFHVPHUIZWCWBVPHVPHIGUOWEUPUQAWDDFURZBAIVOGSZV
+        OUIZCFURWFAWHCFMUSWHWDCDFVOVPUIZWGWCVOVPVOVPIGUOWIUPUQUTVCVAQVBVDVEVFVG
+        $.
+    $}
+
+    $( Deduce right identity from left inverse and left identity in an
+       associative structure (such as a group).  (Contributed by NM,
```

Comment:
Proposal:

> In a semigroup with a left identity and left inverses, that left identity is also a right identity.

Reply:
See above.

Reply:
@benjub The remaining remarks are about the theorems which I moved from subsection "Operations" into PART 10  BASIC ALGEBRAIC STRUCTURES. I would like not to change the comments, as described before. Therefore, I think this PR is ready to be merged.